apps/couchbase.go

```diff
@@ -0,0 +1,179 @@
+package apps
+
+import (
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+// LoggingReceiverCouchbase is a struct used for generating the fluentbit component for couchbase logs
+type LoggingReceiverCouchbase struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the string identifier for the general couchbase logs
+func (lr LoggingReceiverCouchbase) Type() string {
+	return "couchbase_general"
+}
+
+// Components returns the logging components of the couchbase access logs
+func (lr LoggingReceiverCouchbase) Components(tag string) []fluentbit.Component {
+	if len(lr.IncludePaths) == 0 {
+		lr.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/couchdb.log",
+			"/opt/couchbase/var/lib/couchbase/logs/info.log",
+			"/opt/couchbase/var/lib/couchbase/logs/debug.log",
+			"/opt/couchbase/var/lib/couchbase/logs/error.log",
+			"/opt/couchbase/var/lib/couchbase/logs/babysitter.log",
+		}
+	}
+	lr.MultilineRules = []confgenerator.MultilineRule{
+		{
+			StateName: "start_state",
+			NextState: "cont",
+			Regex:     `^\[([^\s+:]*):`,
+		},
+		{
+			StateName: "cont",
+			NextState: "cont",
+			Regex:     `^(?!\[([^\s+:]*):).*$`,
+		},
+	}
+	components := lr.LoggingReceiverFilesMixin.Components(tag)
+	components = append(components, confgenerator.LoggingProcessorParseRegex{
+		Regex: `^\[(?<type>[^:]*):(?<severity>[^,]*),(?<timestamp>\d+-\d+-\d+T\d+:\d+:\d+.\d+Z),(?<node_name>[^:]*):(?<module_name>[^\<]+)(?<source>[^\]]+)\](?<message>.*)$`,
+		ParserShared: confgenerator.ParserShared{
+			TimeKey:    "timestamp",
+			TimeFormat: "%Y-%m-%dT%H:%M:%S.%L",
+		},
+	}.Components(tag, "couchbase_default")...)
+
+	components = append(components,
```

Comment:
Can we use the `modify_fields` processor to update severity. We also need to populate the instrumentation source using the same processor.

https://github.com/GoogleCloudPlatform/ops-agent/blob/bc1e111e672eb17f33b39986924f471938bf09d6/apps/flink.go#L66-L89